File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        Queue<Integer> queue = new LinkedList<>();
        
        for(int i=0; i<arr.length; i++){
            if(stack.isEmpty() || stack.peek() != arr[i]){
                stack.push(arr[i]);
                queue.offer(arr[i]);
            }
        }
        
        int[] answer = new int[queue.size()];
        int index=0;
        
        while(!queue.isEmpty()){
            answer[index++] = queue.poll();
        }

        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    var answer = [];
    
    if(answer.length == 0){
        answer.push(arr[0]);       
    }

    for(var i=1; i<arr.length; i++){
        if(arr[i] != answer[answer.length-1]){
            answer.push(arr[i]);
        }
    }
    
    return answer;
}